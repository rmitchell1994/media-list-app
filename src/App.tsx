import React, { Suspense } from "react";
import Layout from "./atomic/organisms/layout/layout";
import { QueryClient, QueryClientProvider } from "react-query";
import { ReactQueryDevtools } from "react-query/devtools";
import firebase from "firebase/app";
import {
  ThemeProvider,
  createGlobalStyle,
  DefaultTheme,
} from "styled-components";
import { BrowserRouter as Router } from "react-router-dom";
import Routes from "./router";
import theme from "../config/theme.json";
import SignedInProvider from "./context/signed-in-provider";

firebase.initializeApp({
  apiKey: "AIzaSyCoxTs8fCDpVYS2EH9PFqPMYckuIgmooIY",
  authDomain: "media-list-app.firebaseapp.com",
  projectId: "media-list-app",
  storageBucket: "media-list-app.appspot.com",
  messagingSenderId: "622664374317",
  appId: "1:622664374317:web:c5711b9a9e6bbb946e00a4",
  measurementId: "G-XJM9V52B2V",
});

const GlobalStyle = createGlobalStyle`
  body {
    margin: 0;
    padding: 0;
  }

  #root {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }
`;

const queryClient = new QueryClient();

const App: React.FC = () => {
  return (
    <QueryClientProvider client={queryClient}>
      <ThemeProvider theme={theme as DefaultTheme}>
        <SignedInProvider>
          <GlobalStyle />
          <ReactQueryDevtools initialIsOpen />
          <Suspense fallback={"Loading"}>
            <Router>
              <Layout>
                <Routes />
              </Layout>
            </Router>
          </Suspense>
        </SignedInProvider>
      </ThemeProvider>
    </QueryClientProvider>
  );
};

export default App;
