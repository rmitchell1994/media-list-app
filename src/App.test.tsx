import React from "react";
import App from "./App";
import { render } from "@testing-library/react";

describe("App", () => {
  it("should render default component", () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });
});
