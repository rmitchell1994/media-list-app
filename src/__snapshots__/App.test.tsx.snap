// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render default component 1`] = `
<DocumentFragment>
  <div
    class="sc-dlfnbm hUmsUk"
    style="display: none;"
  >
    <header
      class="sc-bdfBwQ etFSeW"
    >
      Media App Header
    </header>
    <div
      class="sc-hKgILt dhzLxW"
    />
    <footer
      class="sc-gsTCUz kRYLqh"
    >
      Footer
    </footer>
  </div>
  Loading
</DocumentFragment>
`;
